scripts/install-mender.sh

```diff
@@ -0,0 +1,144 @@
+ _ __ ___   ___ _ __   __| | ___ _ __ 
+| '_ \` _ \ / _ \ '_ \ / _\` |/ _ \ '__|
+| | | | | |  __/ | | | (_| |  __/ |   
+|_| |_| |_|\___|_| |_|\__,_|\___|_|   
+
+Running the Mender installation script.
+--
+"
+                                      
+}
+
+usage() {
+    echo "usage: install-mender.sh [-h] [-c channel] [component...]"
+    echo "  -h             print this help"
+    echo "  -c CHANNEL     channel: stable(default)|experimental"
+    echo "  <component>    list of components to install"
+    echo ""
+    echo "Supported components (all installed by default):"
+    for c in $DEFAULT_COMPONENTS
+    do
+        echo "  * $c"
+    done
+}
+
+is_known_component() {
+    for known in $DEFAULT_COMPONENTS
+    do
+        if [ "$1" = "$known" ]; then
+            return 0 
+        fi
+    done
+    return 1
+}
+
+parse_args() {
+    selected_components=""
+    while [ $# -gt 0 ]
+    do
+        case $1 in
+            -h)
+                usage
+                exit 0
+                ;;
+            -c)
+                if [ -n "$2" ]; then
+                    CHANNEL=$2
+                    shift 2
+                else
+                    echo "ERROR: channel requires a non-empty option argument."
+                    echo "Aborting."
+                    exit 1
+                fi
+                ;;
+            *)
+                if is_known_component "$1"; then
+                    selected_components="$selected_components $1 "
+                    shift
+                else
+                    echo "Unsupported argument: \`$1\`"
+                    echo "Run \`mender-install.sh -h\` for help."
+                    echo "Aborting."
+                    exit 1
+                fi
+                ;;
+        esac
+    done
+    if [ -n "$selected_components" ]; then
+        SELECTED_COMPONENTS="$selected_components"
+    fi
+}
+
+print_components() {
+    echo "  Selected components:"
+    for c in $SELECTED_COMPONENTS
+    do
+        printf "\t%s\n" "$c"
+    done
+}
+
+init() {
+    REPO_URL=https://downloads.mender.io/repos/debian
+
+    parse_args "$@"
+
+    ARCH=$(dpkg --print-architecture)
+    echo "  Detected architecture:"
+    printf "\t%s\n" "$ARCH"
+
+    echo "  Installing from channel:"
+    printf "\t%s\n" "$CHANNEL"
+}
+
+get_deps() {
+    apt-get update -qq
+    apt-get install -y -qq --no-install-recommends \
+        apt-transport-https \
+        ca-certificates \
+        curl \
+        gnupg-agent \
+        software-properties-common
+}
+
+add_repo() {
+    curl -fsSL $REPO_URL/gpg | apt-key add -
+
+    add-apt-repository \
+        "deb [arch=$ARCH] $REPO_URL \
+        $CHANNEL \
+        main"
+}
+
+inst_component() {
+    echo "Installing $1"
+}
+
+do_install() {
+    apt-get update 
+    apt-get install -y $SELECTED_COMPONENTS
+
+    echo "  Success!"
+    echo "  Please run \`mender setup\` to configure the client."
+    exit 0
+}
+
+banner
+init "$@"
+print_components
+get_deps
+add_repo
```

Comment:
It came to my mind now, specifically after reviewing the docs PR, that we should check if the repo is already installed before installing it. See [this comment](https://github.com/mendersoftware/mender-docs/pull/1246#discussion_r531145192).

It wasn't part of the acceptance criteria for the script to be idempotent, but it makes sense to me. Is this something easy to check?

Reply:
oops, I only just noticed your remark, but yes, it should be fairly easy.